### PR TITLE
hipblasLtMatmul supports nullptr

### PR DESCRIPTION
**Pull request by wenchuanchen**
The request is from https://ontrack-internal.amd.com/browse/SWDEV-428339

### CODE REVIEW

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 1]
```diff
@@ -1192,6 +1206,23 @@ rocblaslt_status runContractionProblem(rocblaslt_handle                   handle
         hardware     = Tensile::hip::GetDevice(*deviceProp);
 
         std::shared_ptr<TensileDataGemm> data = std::static_pointer_cast<TensileDataGemm>(gemmData);
+        int requestedAlgoCount = 1, returnAlgoCount;
+        rocblaslt_matmul_heuristic_result heuristicResultsArray[requestedAlgoCount];
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: unnessary to allocate array.

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 2]
```diff
@@ -1192,6 +1206,23 @@ rocblaslt_status runContractionProblem(rocblaslt_handle                   handle
...
+        if (algo == nullptr) {
+            status = getBestSolutions(prob,
+                                      handle,
+                                      gemmData,
+                                      requestedAlgoCount,
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: setting 1 here should be enough

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 3]
```diff
@@ -1192,6 +1206,23 @@ rocblaslt_status runContractionProblem(rocblaslt_handle                   handle
...
+                                      heuristicResultsArray,
+                                      &returnAlgoCount,
+                                      prob.workspaceSize);
+            if (returnAlgoCount == 0) return rocblaslt_status_not_implemented;
+            algo = &heuristicResultsArray[0].algo;
+            if (prob.workspaceSize == 0 || prob.workspace == nullptr) {
+                prob.workspaceSize = heuristicResultsArray[0].workspaceSize;
+                CHECK_HIP_ERROR(hipMalloc(&prob.workspace, prob.workspaceSize));
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: don't allocate workspace in hipblaslt runtime.
getBestSolutions already has the workspace size info to get the proper solution

File: clients/include/testing_auxiliary.hpp [hunk 4]
```diff
@@ -304,6 +304,94 @@ void testing_aux_matmul_alg_set_attr_bad_arg(const Arguments& arg) {}
 
 void testing_aux_matmul_alg_get_attr_bad_arg(const Arguments& arg) {}
 
+void testing_aux_matmul_alg_matmul_bad_arg(const Arguments& arg)
```
Comment (jichangjichang) [inline clients/include/testing_auxiliary.hpp]: "bad_arg" suffix means that you expect some error code should be returned in certain function.
The below codes didn't have that checking.

File: clients/gtest/auxiliary_gtest.yaml [hunk 5]
```diff
@@ -99,4 +99,13 @@ Tests:
   function:
     - aux_matmul_pref_init: *real_precisions
 
+- name: aux_matmul_alg_matmul_bad_arg
+  category: pre_checkin
+  function:
+    - aux_matmul_alg_matmul_bad_arg: *real_precisions
```
Comment (jichangjichang) [inline clients/gtest/auxiliary_gtest.yaml]: using hpa_half_precision if you want hard-code inside the test function.

**Review by KKyang — APPROVED** [review]

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 6]
```diff
@@ -1192,6 +1192,19 @@ rocblaslt_status runContractionProblem(rocblaslt_handle                   handle
         hardware     = Tensile::hip::GetDevice(*deviceProp);
 
         std::shared_ptr<TensileDataGemm> data = std::static_pointer_cast<TensileDataGemm>(gemmData);
+        rocblaslt_matmul_heuristic_result heuristicResult;
+        int returnAlgoCount;
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: move into if (algo == nullptr){}

**Review by jichangjichang — APPROVED** [review]
lgtm

**Pull request merged**